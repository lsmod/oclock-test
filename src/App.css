.App {
  text-align: center;
  background-color: #402e09;
  background-image: url(assets/denys-nevozhai-j8Bc4UerZXY-unsplash-reduced.jpg);
  background-size: cover;
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-link {
  color: #61dafb;
}
