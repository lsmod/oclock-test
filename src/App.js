import React from "react";
import "./App.css";
import CurrencyConverter from "./components/CurrencyConverter";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Convertiseur de devises</h1>
        <CurrencyConverter />
      </header>
    </div>
  );
}

export default App;
