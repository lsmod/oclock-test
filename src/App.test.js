import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

describe("Test composant <App />", () => {
  it("le titre est bien 'Convertisseur de devices'", () => {
    const { getByText, asFragment } = render(<App />);
    const linkElement = getByText("Convertiseur de devises");
    expect(linkElement).toBeInTheDocument();
  });

  it("le titre est un <h1>", () => {
    const { getByText, asFragment } = render(<App />);
    const titleH1 = document.querySelector("h1");
    expect(titleH1).toBeInTheDocument();
  });

  it("le titre est imbriqué dans un <header>", () => {
    const { getByText, asFragment } = render(<App />);
    const header = document.querySelector("header");
    const titleH1 = document.querySelector("h1");
    expect(header).toContainElement(titleH1);
  });

  it("le header a bien la classe css 'App-header'", () => {
    const { getByText, asFragment } = render(<App />);
    const header = document.querySelector("header.App-header");
    expect(header).toBeInTheDocument(header);
  });

  it("les snapshot", () => {
    const { asFragment } = render(<App />);
    // à chaque fois que les tests sont lancés une "snapshot" du dom est prise (sauvegarde du code html produit)
    // cette snapshot est sauvegardée dans un fichier .snap et comparé au prochain test pour véfifier que le html est toujours le même
    expect(asFragment()).toMatchSnapshot();
  });
});
