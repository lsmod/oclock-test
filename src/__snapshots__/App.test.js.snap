// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test composant <App /> les snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2>
            Convertisseur de devises
          </h2>
          <div>
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
            <br />
            Chargement des taux de change
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
